### PR TITLE
Add classification docs

### PR DESCRIPTION
**Pull request by jmattick**

### CODE REVIEW

File: docs/classification/pigeon-output.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+nav_order: 5
+---
+
+## Pigeon _classify_ and _filter_ output
+
+### Classification File
+
+The _classify_ and _filter_ tools output a txt file containing isoform annotation information.
+The output from _classify_ and _filter_ will have the extentions `_classification.txt` or `_classification.filtered_lite_classification.txt` respectively.
+Both of these outputs follow the SQANTI3 [classification file](https://github.com/ConesaLab/SQANTI3/wiki/Understanding-the-output-of-SQANTI3-QC#glossary-of-classification-file-columns-classificationtxt) convention with the expection of two added columns.
```
Comment (pezmaster31) [inline docs/classification/pigeon-output.md]: exception

File: docs/classification/pigeon-output.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+title: Pigeon Output
+nav_order: 5
+---
+
+## Pigeon _classify_ and _filter_ output
+
+### Classification File
+
+The _classify_ and _filter_ tools output a txt file containing isoform annotation information.
+The output from _classify_ and _filter_ will have the extentions `_classification.txt` or `_classification.filtered_lite_classification.txt` respectively.
```
Comment (pezmaster31) [inline docs/classification/pigeon-output.md]: extensions

File: docs/classification/workflow.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+Sort the transcript GFF file output from _isoseq collapse_.
+
+```
+pigeon sort <collapse.gff> -o sorted.gff
+```
+
+### Index the genome annotation
+
+Index the genome annotation file before classification.
```
Comment (pezmaster31) [inline docs/classification/workflow.md]: I would mention that the supplemental annotation files (CAGE & intropolis) need to be pigeon-indexed as well.

**Review by pezmaster31 — APPROVED** [review]

File: docs/classification/isoseq-collapse.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+Collapse mapped reads into unique isoforms using _isoseq collapse_.
+
+```
+isoseq3 collapse <mapped.bam> <collapse.gff>
+```
+
+### Ouptut
+
+- `collapse.gff` contains the collapsed isoforms in gff format.
+- `*.abundance.txt` contains information about the number of FLNC reads supporting each isoform and cell barcodes if applicable.
```
Comment (Magdoll) [inline docs/classification/isoseq-collapse.md]: add:
"each unique isoform has the ID format PB.X.Y, while `count_fl` denotes the number of unique molecules (after UMI deduplication) supporting the isoform, and `fl_assoc` denotes the number of reads (before UMI deduplication) supporting it. `cell_barcodes` shows the list of single cell barcodes from which the reads came from, if applicable.

File: docs/classification/isoseq-collapse.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    ```
+    pbid	count_fl	fl_assoc	cell_barcodes
+    PB.1.1	2	2	ATCCATTCACCTCTGT,ATCGGCGCAGAGATGC
+    PB.2.1	1	1	CGGACACCATTGCCGG
+    PB.3.1	1	1	ACTTCGCGTCTAACTG
+    ```
+- `*.group.txt` contains the grouped input by isoform.
```
Comment (Magdoll) [inline docs/classification/isoseq-collapse.md]: edit to: "...shows the grouping of redundant isoforms (based on mapped exonic structures), where the read names `molecule/<number>` denote a unique molecule after UMI deduplication."

File: docs/classification/isoseq-collapse.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    ```
+    PB.1.1	molecule/7343975,molecule/7738347
+    PB.2.1	molecule/14601188
+    PB.3.1	molecule/3998518
+    ```
+- `*.read_stat.txt` contains the isoforms that reads are assigned to.
```
Comment (Magdoll) [inline docs/classification/isoseq-collapse.md]: edit to: "...shows the assignment of each read (before UMI deduplication) to the final, unique isoforms PB.X.Y. Read names with the format `<movie>/<zmw>/ccs` indicate a CCS read, whereas `<movie>/<zmw>/ccs/<start>_<end>` further denotes a segment of a CCS read (S-read), likely as a result of segmentation (using, for example, [Skera](https://skera.how)) of concatenated single cell libraries.

File: docs/classification/workflow.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+Sort the transcript GFF file output from _isoseq collapse_.
+
+```
+pigeon sort <collapse.gff> -o sorted.gff
+```
+
+### Index the reference files
+
+Index the genome annotation, CAGE peak, and intropolis files before classification.
```
Comment (Magdoll) [inline docs/classification/workflow.md]: if CAGE peak and intropolis is optional, note "(optional)" in front.

File: docs/umi/examples.md [hunk 8]
```diff
@@ -15,13 +15,13 @@ This is an example of an end-to-end cmd-line-only workflow:
...
-    lima 2.0.0 (commit v2.0.0)
+    lima 2.6.0 (commit v2.6.0)
 
-    # Check isoseq3 version to be >= 3.4.0
+    # Check isoseq3 version to be >= 3.7.0
     $ isoseq3 --version
-    isoseq3 3.4.0 (commit v3.4.0)
+    isoseq3 3.7.0 (commit v3.7.0)
 
     # Primer removal
```
Comment (Magdoll) [inline docs/umi/examples.md]: Change to "cDNA primer removal and read orientation"

File: docs/umi/examples.md [hunk 9]
```diff
@@ -32,5 +32,10 @@ This is an example of an end-to-end cmd-line-only workflow:
     # Remove poly(A) tails and concatemer
     $ isoseq3 refine flt.bam primers.fasta fltnc.bam --require-polya
 
+    # Correct Barcodes
```
Comment (Magdoll) [inline docs/umi/examples.md]: change comment to "Correct single cell barcodes based on an include list"

File: docs/umi/examples.md [hunk 10]
```diff
@@ -32,5 +32,10 @@ This is an example of an end-to-end cmd-line-only workflow:
...
+    $ isoseq3 correct --barcodes include.txt fltnc.bam corrected.bam
+
+    $samtools sort -t CB corrected.bam -o corrected.sorted.bam
+
     # Deduplicate transcripts from the identical molecule
```
Comment (Magdoll) [inline docs/umi/examples.md]: we aren't deduplicating transcripts - we are deduplicating identical molecules (you have still end up w unique molecules representing the same transcript), hence change to

"# Deduplicate reads based on UMIs"

File: docs/umi/examples.md [hunk 11]
```diff
@@ -32,5 +32,10 @@ This is an example of an end-to-end cmd-line-only workflow:
     # Remove poly(A) tails and concatemer
     $ isoseq3 refine flt.bam primers.fasta fltnc.bam --require-polya
 
+    # Correct Barcodes
+    $ isoseq3 correct --barcodes include.txt fltnc.bam corrected.bam
+
+    $samtools sort -t CB corrected.bam -o corrected.sorted.bam
```
Comment (Magdoll) [inline docs/umi/examples.md]: Move this to after the "#deduplicate" header and right before "isoseq3 groupdedup" command

File: docs/classification/workflow.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+### Map reads to a reference genome
+
+Map reads using _pbmm2_.
+
+```
+pbmm2 align --preset ISOSEQ --sort <input.bam> <ref.fa> <mapped.bam>
+```
+
+### Collapse transcripts into unique isoforms
```
Comment (Magdoll) [inline docs/classification/workflow.md]: change to "### Collapse into unique isoforms"

File: docs/classification/workflow.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+
+Collapse mapped reads into unique isoforms using _isoseq collapse_.
```
Comment (Magdoll) [inline docs/classification/workflow.md]: change to "collapse redundant transcripts into unique isoforms based on exonic structures using _isoseq collapse_."

File: docs/classification/workflow.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+
+```
+isoseq3 collapse <mapped.bam> <collapse.gff>
```
Comment (Magdoll) [inline docs/classification/workflow.md]: collapse --> collapsed

File: docs/classification/workflow.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+```
+
+### Sort input transcript GFF
+
+Sort the transcript GFF file output from _isoseq collapse_.
+
+```
+pigeon sort <collapse.gff> -o sorted.gff
```
Comment (Magdoll) [inline docs/classification/workflow.md]: collapse.gff --> collapsed.gff

File: docs/classification/isoseq-collapse.md [hunk 16]
```diff
@@ -0,0 +1,53 @@
+---
+layout: default
+parent: Classification
+title: Pigeon Workflow
+nav_order: 2
+---
+
+## IsoSeq collapse
+
+After reads are mapped to a reference genome, transcripts can be collapsed into unique isoforms using `isoseq3 collapse`. Mapped IsoSeq reads are collapsed to produce a set of unique isoforms for each locus of the genome in GFF format, along with a secondary files containing information about the number of processed reads supporting each unique isoform.
```
Comment (Magdoll) [inline docs/classification/isoseq-collapse.md]: "after transcript sequences are mapped to a reference genome, `isoseq3 collapse` can be used to collapse redundant transcripts (based on exonic structures) into unique isoforms. Output consists of unique isoforms in GFF format and secondary files containing information about the number of reads supporting each unique isoform"

**Pull request merged**